cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -285,11 +376,22 @@ class ndview : public ndarray_base<axis_count, order> {
     }
 
 #ifdef ONEDAL_DATA_PARALLEL
+    /// Returns a copy of the element of 1d immutable view located in USM at specified location.
```

Comment:
Does it execute operations in a queue if the data depends on it?

Reply:
I think I do not fully understand the question. This operation is about the same as other sycl operations, it will be included into the flow of dependencies along with the other operations.

Reply:
Suppose that the array is the result of a computation like $t(X)*X$.

The operation that calculates the numbers in the array returns a queue, and might not have been completed by the time that `.at` is called.

Will the call to `.at` force calculations of the operations in the queues before returning a scalar?

Reply:
Sure, in case the events triggered by $t(X) \cdot X$ are passed into the `deps` vector correctly.